dask_remote/runner/runner.py

```diff
@@ -0,0 +1,96 @@
+"""
+Run Cluster in a separata process, and expose its scaling commands through a "proxy".
+
+>>> from multiprocessing import Pipe
+>>> from dask.distributed import LocalCluster
+>>> from dask_remote import ClusterProcess, ClusterProcessProxy
+>>> cmd_conn, result_conn = Pipe()
+>>> cluster_proc = ClusterProcess(cmd_conn, result_conn, LocalCluster, dict(n_workers=0))
+>>> cluster_proc.start()  # uses the multiprocessing.Process API
+>>> cluster_proxy = ClusterProcessProxy(cmd_conn, result_conn)
+>>> cluster_proxy.scale(4)  # command is proxied to the cluster object in the child process
+>>> cluster_prox.join()  # cluster remains alive until terminated
+"""
+from multiprocessing import Process
+from multiprocessing.connection import Connection
+
+
+class ClusterProcess(Process):
+    """Run a dask Cluster object in a child process, and expose core methods and attributes.
+    """
+
+    def __init__(
+        self, cmd_conn: Connection, result_conn: Connection, cluster_cls, cluster_kwargs=None
+    ):
+        self.cluster_cls = cluster_cls
+        self.cluster_kwargs = cluster_kwargs or {}
+        self.cmd_conn = cmd_conn  # pipe connection to receive scaling/control commands from
+        self.result_conn = result_conn  # pipe connection to return messages to
+        super().__init__()
+
+    # Process side
+    def run(self):
+        cluster = self.cluster_cls(**self.cluster_kwargs)
+        while True:
+            cmd = self.cmd_conn.recv()
+            self._process_cmd(cmd, cluster)
+
+    def _process_cmd(self, cmd, cluster):
+        try:
+            result = self._call_cmd(cmd, cluster)
+        except Exception as e:
+            result = e
+        self.result_conn.send(result)
+
+    @staticmethod
+    def _call_cmd(cmd, obj):
+        if "method" in cmd:
+            method = cmd["method"]
+            args = cmd.get("args", [])
+            kwargs = cmd.get("kwargs", {})
+            return getattr(obj, method)(*args, **kwargs)
+        elif "attribute" in cmd:
+            attribute = cmd["attribute"]
+            return getattr(obj, attribute)
+
+
+class ClusterProcessProxy:
+
+    CLUSTER_ATTRIBUTES = ["dashboard_link", "scheduler_address", "scheduler_info"]
+    CLUSTER_METHODS = ["scale", "adapt"]
```

Comment:
not that is super pressing, but maybe inheriting from whichever class those methods/properties are defined would make this a bit more tight.

Reply:
Yes, I have thought about it but (a) class introspection is hard, but more importantly (b) I am constraining to methods that I know return something serializable